evalml/tests/pipeline_tests/test_pipeline_utils.py

```diff
@@ -0,0 +1,478 @@
+def test_stacked_estimator_in_pipeline(problem_type, X_y_binary, X_y_multi, X_y_regression,
+                                       stackable_classifiers,
+                                       stackable_regressors,
+                                       logistic_regression_binary_pipeline_class,
+                                       logistic_regression_multiclass_pipeline_class,
+                                       linear_regression_pipeline_class):
+    if problem_type == ProblemTypes.BINARY:
+        X, y = X_y_binary
+        base_pipeline_class = BinaryClassificationPipeline
+        stacking_component_name = StackedEnsembleClassifier.name
+        input_pipelines = [make_pipeline_from_components([classifier], problem_type) for classifier in stackable_classifiers]
+        comparison_pipeline = logistic_regression_binary_pipeline_class(parameters={"Logistic Regression Classifier": {"n_jobs": 1}})
+        objective = 'Log Loss Binary'
+    elif problem_type == ProblemTypes.MULTICLASS:
+        X, y = X_y_multi
+        base_pipeline_class = MulticlassClassificationPipeline
+        stacking_component_name = StackedEnsembleClassifier.name
+        input_pipelines = [make_pipeline_from_components([classifier], problem_type) for classifier in stackable_classifiers]
+        comparison_pipeline = logistic_regression_multiclass_pipeline_class(parameters={"Logistic Regression Classifier": {"n_jobs": 1}})
+        objective = 'Log Loss Multiclass'
+    elif problem_type == ProblemTypes.REGRESSION:
+        X, y = X_y_regression
+        base_pipeline_class = RegressionPipeline
+        stacking_component_name = StackedEnsembleRegressor.name
+        input_pipelines = [make_pipeline_from_components([regressor], problem_type) for regressor in stackable_regressors]
+        comparison_pipeline = linear_regression_pipeline_class(parameters={"Linear Regressor": {"n_jobs": 1}})
+        objective = 'R2'
+    parameters = {
+        stacking_component_name: {
+            "input_pipelines": input_pipelines,
+            "n_jobs": 1
+        }
+    }
+    graph = ['Simple Imputer', stacking_component_name]
+
+    class StackedPipeline(base_pipeline_class):
+        component_graph = graph
+        model_family = ModelFamily.ENSEMBLE
+
+    pipeline = StackedPipeline(parameters=parameters)
+    pipeline.fit(X, y)
+    comparison_pipeline.fit(X, y)
+    assert not np.isnan(pipeline.predict(X).to_series()).values.any()
+
+    pipeline_score = pipeline.score(X, y, [objective])[objective]
+    comparison_pipeline_score = comparison_pipeline.score(X, y, [objective])[objective]
+
+    if problem_type == ProblemTypes.BINARY or problem_type == ProblemTypes.MULTICLASS:
+        assert not np.isnan(pipeline.predict_proba(X).to_dataframe()).values.any()
+        assert (pipeline_score <= comparison_pipeline_score)
+    else:
+        assert (pipeline_score >= comparison_pipeline_score)
```

Comment:
Moved from `test_pipelines.py` since these are stored in `evalml/pipelines/utils.py` and `test_pipelines.py` is huge